### PR TITLE
feat: add collections endpoints

### PR DESCRIPTION
**Pull request by p4czyk**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Issues Fixed 
* [Issue #202]

### Changes Proposed

* Add support for the collections endpoints in the Kraken API.

### Additional Information
* Request methods with auth token will be tested by @iProdigy.

### CODE REVIEW

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java:19]:
```suggestion
    private Instant createdAt;
```

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java:18]: You can avoid this via @JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class) (applies for numerous other fields in this class too)

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java:47]:
```suggestion
    private Instant updatedAt;
```

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java:11]:
```suggestion
@Data
@Setter(AccessLevel.PRIVATE)
```

**Review by iProdigy — CHANGES_REQUESTED** [review]
Thank you for the PR!

Among the minor changes:
* variables with `ID` should be `Id` to conform to lowerCamelCase
* token param was missing from javadocs
* can utilize `@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)` to avoid so much manual `@JsonProperty` use

**Comment by stachu540** [comment]
> * can utilize `@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)` to avoid so much manual `@JsonProperty` use

do suggest add [here](https://github.com/twitch4j/twitch4j/blob/350c8b9ff34d323313a4609611f984f5dcda48ae/common/src/main/java/com/github/twitch4j/common/util/TypeConvert.java#L15-L17i)
```java
setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
```

**Comment by iProdigy** [comment]
@stachu540 Good point. I'll include that in #231 though

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java:296]:
```suggestion
        "Accept: application/vnd.twitchtv.v5+json",
        "Content-Type: application/json"
```

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java:337]:
```suggestion
        "Accept: application/vnd.twitchtv.v5+json",
        "Content-Type: application/json"
```

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java
Comment (iProdigy) [inline rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKraken.java:381]:
```suggestion
        "Accept: application/vnd.twitchtv.v5+json",
        "Content-Type: application/json"
```

**Review by iProdigy — COMMENTED** [review]
* JSON `@Body` requires `Content-Type` header
* Fix return type of addItemToCollection
* Fix json body for addItemToCollection

**Comment by stachu540** [comment]
I have another suggestions after [this line](https://github.com/twitch4j/twitch4j/blob/7d7574f3fe98d0423eb34b3d2e615375c68daaba/rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKrakenBuilder.java#L120)
```java
.requestInterceptor(t -> {
  t.header("Accept", "application/vnd.twitchtv.v5+json");
  t.header("Content-Type", "application/json");
});
```

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java
Comment (p4czyk) [reply rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollectionMetadata.java:19]: Should I apply this to the classes or will it be fixed in another pr?

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java
Comment (p4czyk) [reply rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java]: Applied in cd4e0e2.

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java
Comment (p4czyk) [reply rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java]: Applied in 807bdc0.

File: rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java
Comment (p4czyk) [reply rest-kraken/src/main/java/com/github/twitch4j/kraken/domain/KrakenCollection.java]: Applied in 807bdc0.

**Review by p4czyk — COMMENTED** [review]
Every request should be applied now. 🚀👍

**Review by iProdigy — APPROVED** [review]
Thank you for your contribution!

**Review by PhilippHeuer — APPROVED** [review]

**Comment by PhilippHeuer** [comment]
> I have another suggestions after [this line](https://github.com/twitch4j/twitch4j/blob/7d7574f3fe98d0423eb34b3d2e615375c68daaba/rest-kraken/src/main/java/com/github/twitch4j/kraken/TwitchKrakenBuilder.java#L120)
> 
> ```java
> .requestInterceptor(t -> {
>   t.header("Accept", "application/vnd.twitchtv.v5+json");
>   t.header("Content-Type", "application/json");
> });
> ```

👍  good suggestion, i will make this change in a new pr after we merge this one (unless you want to @stachu540 ) - it still needs a if to only apply content-type for post/put/patch requests.

**Pull request merged**
